sklearn/naive_bayes.py

```diff
@@ -1154,25 +1154,33 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
 
     def _check_X(self, X):
         # FIXME: we can avoid calling check_array twice after #14872 is merged.
-        # return check_array(X, y, dtype='int', accept_sparse=False,
-        #                   force_all_finite=True)
+        # X = check_array(X, y, dtype='int', accept_sparse=False,
+        #                 force_all_finite=True)
         X = check_array(X, accept_sparse=False, force_all_finite=True)
-        return check_array(X, dtype='int')
+        X = check_array(X, dtype='int')
+        return self._check_nonnegative(X)
 
     def _check_X_y(self, X, y):
         # FIXME: we can avoid calling check_array twice after #14872 is merged.
-        # return check_array(X, y, dtype='int', accept_sparse=False,
-        #                   force_all_finite=True)
+        # X, y = check_array(X, y, dtype='int', accept_sparse=False,
+        #                    force_all_finite=True)
         X, y = check_X_y(X, y, accept_sparse=False, force_all_finite=True)
-        return check_X_y(X, y, dtype='int')
+        X, y = check_X_y(X, y, dtype='int')
+        return self._check_nonnegative(X), y
```

Comment:
Please just inline this. I think `if any(X < 0)` followed by an error message is about as explicit as it gets without needing a function name